app/components/Song.tsx

```
@@ -0,0 +1,41 @@
+import React from 'react';
+import { type CurrentMusic, type Song } from '~/types';
+
+type SongOrCurrentMusic = Song | CurrentMusic;
+
+const isCurrentMusic = (song: SongOrCurrentMusic): song is CurrentMusic => {
+  return (song as CurrentMusic).isPlaying !== undefined;
+};
+
+const QualityBadge: React.FC<{ media?: SongOrCurrentMusic["Media"] }> = ({ media }) => {
+  let quality = 'Lossy';
+  if (media && media[0]) {
```

Biome

[error] Change to an optional chain.

Unsafe fix: Change to an optional chain.


(lint/complexity/useOptionalChain)